src/app/etl-api/surge-resource.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { AppSettingsService } from '../app-settings/app-settings.service';
+import { catchError, map } from 'rxjs/operators';
+import * as Moment from 'moment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SurgeResourceService {
+  public get url(): string {
+    return this.appSettingsService.getEtlRestbaseurl().trim();
+  }
+
+  constructor(
+    public http: HttpClient,
+    public appSettingsService: AppSettingsService
+  ) { }
+
+  public getSurgeWeeklyReport(params: any): Observable<any> {
+    // tslint:disable-next-line: max-line-length
+    return this.http.get(`${this.url}surge-report?year_week=${Moment(params.year_week).format('YYYYWW')}&locationUuids=${params.locationUuids}`)
+      .pipe(
+        catchError((err: any) => {
+          const error: any = err;
+          const errorObj = {
+            error: error.status,
+            message: error.statusText
+          };
+          return Observable.of(errorObj);
+        }),
+        map((response: Response) => {
+          return response;
+        })
+      );
+  }
+
+  public getSurgePatientList(params: any): Observable<any> {
```

Comment:
rename to getSurgeWeeklyPatientList or if you plan to re-use it to get the daily report, then rename line 21 appropriately